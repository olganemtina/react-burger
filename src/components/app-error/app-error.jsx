import PropTypes from 'prop-types';

export default function AppError(props) {
    return (
        <h1 className="text text_type_main-small mt-4 text_color_inactive">{props.error}</h1>
    )
};

AppError.propTypes = {
    error: PropTypes.string.isRequired,
};

