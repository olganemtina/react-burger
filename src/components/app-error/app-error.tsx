import { FC } from 'react';

export const AppError:FC<{error: string}> = ({error}) => {
    return (
        <h1 className="text text_type_main-small mt-4 text_color_inactive">{error}</h1>
    )
};

