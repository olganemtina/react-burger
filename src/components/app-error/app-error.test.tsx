import React from "react";
import renderer from "react-test-renderer";
import { shallow, configure } from "enzyme";
import { AppError } from "./app-error";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("Компонент без ошибок", () => {
	const tree = renderer.create(<AppError error="Тест" />).toJSON();
	expect(tree).toMatchSnapshot();
});
