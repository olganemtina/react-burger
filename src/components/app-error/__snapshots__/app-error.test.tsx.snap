// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент без ошибок 1`] = `
<h1
  className="text text_type_main-small mt-4 text_color_inactive"
>
  Тест
</h1>
`;
